tornado/netutil.py

```diff
@@ -48,15 +48,23 @@
 # For undiagnosed reasons, 'latin1' codec may also need to be preloaded.
 u"foo".encode("latin1")
 
-# Default backlog used when calling sock.listen()
-_DEFAULT_BACKLOG = 128
+# Default backlog used when calling sock.listen(); if `None`, no value is
+# passed to `socket.listen` implying normally that Python will decide on
+# the backlog size for the socket. For Python < 3.6, an error will be
+# raised by `socket.listen` if no argument is provided.
+DEFAULT_BACKLOG = None
+
+# Number of connections to try and accept in one sweep of an I/O loop;
+# if falsy (e.g. `None` or zero) the number will be decided on
```

Comment:
These comments imply that monkey-patching these constants is intended to be a public interface for configuration. I don't think that's a good idea (or at least I'd want to see some more justification for why someone might want to change this so we could think about the most appropriate interface). 

I'd rather just hard-code the number of iterations in `accept_handler`, and get rid of the tornado-side `_DEFAULT_BACKLOG` (so we call `listen()` with an argument if a backlog is given, and otherwise use no arguments so that the default is determined by lower levels).

Reply:
I am currently debugging a case with our usage of tornado where we do want to change this to a much smaller value. 

What we have seen is that in multiprocessing pre-fork setups, where we have a large number of workers sharing 1 socket, we see very poor tail latencies under high load. We have narrowed down due to huge amounts of queuing within processing that accidentally accept far too high of a share of the incoming connections. IE while we might have on average 3 async requests in flight per worker, when we see the outliers with high tail latencies hitting timeouts and inspect their io_loops we'll see that 30 requests connections got all accepted at once on that worker, and that 30th request queue up at the end in the worker will have massively increased latency. 

We are currently looking at ways to prevent this poor load balancing, and at least one element of it, might be to limit the number of connection accepted at a time, so that if a small queue under high load of pending connections has built up, it's not entirely taken by a single worker, and is better distributed in smaller chunks over multiple workers.

Reply:
Interesting. That does sound like a good reason to make the accept-calls-per-loop value configurable (or maybe just to hard-code it to a small value, but I suspect we'd have a hard time agreeing on what that value should be), so the accept-calls-per-loop variable is valuable even aside from its use here for the case when the accept backlog is unspecified/None.

I still don't like monkey-patching module-level constants as the configuration interface, but it seems like whatever interface we come up with should be used for both variables.

In the meantime, have you tried SO_REUSEPORT (and binding the socket after the fork)? My understanding is that the kernel treats this differently from the case of a socket opened before a fork, in a way that generally does a better job at balancing the load across the processes. I don't have any experience with it at scale, though.